### PR TITLE
Feedback forms submission

### PR DESCRIPTION
**Pull request by shinyichen**
Making feedback form functional
* Load record
* Validate bibcode
* Improve preview
* Add diff view
* Form submission
* Change to `react hook form` : This modified all the forms component a lot, although UI look the same.

### The next round will handle these things
* General feedback: add addition information
   * platform, url, browser, current query, current user, browser
* Missing record form
  * pagination authors and other tables
  * reorder authors

### CODE REVIEW

File: src/api/feedback/feedback.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    cacheTime: 0,
+  });
+};
+
+const feedbackQueryFn: QueryFunction<IADSApiFeedbackResponse> = async ({ meta }) => {
+  const { params } = meta as { params: Partial<IFeedbackParams> };
+
+  const config: ApiRequestConfig = {
+    method: 'POST',
+    url: `${ApiTargets.FEEDBACK}`,
```
Comment (thostetler) [inline src/api/feedback/feedback.ts:24]: Could remove the template

File: src/api/feedback/feedback.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import api, { ApiRequestConfig, ApiTargets } from '@api';
+import { QueryFunction, useQuery } from '@tanstack/react-query';
+import { IFeedbackParams, FeedbackADSQuery, IADSApiFeedbackResponse } from './types';
+
+const feedbackKeys = {
+  general: (params: IFeedbackParams) => ['feedback/general', { params }] as const,
+};
+
+export const useFeedback: FeedbackADSQuery = (params, options) => {
+  return useQuery({
```
Comment (thostetler) [inline src/api/feedback/feedback.ts:10]: I've found that useMutation is better choice for these types of queries.  I'm thinking we'll start to transition over to using that for any non-GET requests in the future

File: src/api/search/search.ts [hunk 3]
```diff
@@ -227,6 +229,20 @@ export const useGetAbstractPreview: SearchADSQuery<{ bibcode: IDocsEntity['bibco
   });
 };
 
+/**
+ * Get a single record using identifier for feedack form
+ */
+export const useGetSingleRecord: SearchADSQuery<{ id: string }> = ({ id }, options) => {
```
Comment (thostetler) [inline src/api/search/search.ts:235]: Nice, I think this makes sense

File: src/components/FeedbackForms/AssociatedArticles/AssociatedArticlesForm.tsx [hunk 4]
```diff
@@ -9,87 +12,266 @@ import {
...
+  relationship: Relationship;
   otherRelationship: string;
   mainBibcode: string;
-  associatedBibcodes: string[];
+  associatedBibcodes: AssociatedBibcode[];
 };
 
+type State = 'idle' | 'submitting' | 'validate-bibcodes' | 'preview';
+
+const validationSchema: Yup.ObjectSchema<FormValues> = Yup.object({
```
Comment (thostetler) [inline src/components/FeedbackForms/AssociatedArticles/AssociatedArticlesForm.tsx:37]: We actually already have `zod` installed

File: src/components/FeedbackForms/util.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import e from 'express';
```
Comment (thostetler) [inline src/components/FeedbackForms/util.ts:1]: bad import?

File: src/components/FeedbackForms/MissingRecord/KeywordsField.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  // enable add keyword if new keyword is valid
+  const isNewKeywordValid = newKeyword.length > 0 && keywords.findIndex((kw) => kw.value === newKeyword) === -1;
+
+  const handleNewKeywordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setNewKeyword(e.target.value);
+  };
+
+  const handleAddKeyword = () => {
+    append({ value: newKeyword });
```
Comment (thostetler) [inline src/components/FeedbackForms/MissingRecord/KeywordsField.tsx:26]: should make sure to send focus back to the input after adding the keyword

**Review by thostetler — APPROVED** [review]
LGTM, just had a few comments

**Comment by thostetler** [comment]
Wasn't able to load a bibcode on the edit record side: 
image

If checking to make sure it's a proper identifier, we should add a message and maybe keep the button enabled

**Comment by shinyichen** [comment]
@thostetler When you get another chance, please take a look at the changes:

The last 6 commits try to resolve the feedbacks from you, and also add :
* author table pagination / reordering
* Extra information to send in general feedback

**Comment by thostetler** [comment]
@shinyichen These changes look good.  The forms seem to work well.

We should have a link to update/fix article on the abstract pages, like in BBB

**Pull request merged**